File: js-jest/src/AgedBrie.js
module.exports = class AgedBrie {
    constructor(item) {
        this.item = item;
    }

    update() {
        this.item.sellIn -= 1;

        const {quality} = this.item;

        if (this.item.sellIn < 0) {
            this.item.quality = quality + 2 > 50 ? 50 : quality + 2;
        } else {
            this.item.quality = quality + 1 > 50 ? 50 : quality + 1;
        }
    }
}

File: go/gildedrose/strategy/agedBrie.go
package strategy

import "github.com/emilybache/gildedrose-refactoring-kata/gildedrose/item"

type AgedBrie struct {
	*item.Item
}

func (item *AgedBrie) UpdateSellIn() {
	item.Item.SellIn -= 1
}

func (item *AgedBrie) UpdateQuality() {
	if item.Item.Quality < 50 {
		if item.Item.SellIn <= 0 {
			item.Item.Quality += 2
		} else {
			item.Item.Quality += 1
		}
	}
	if item.Item.Quality > 50 {
		item.Item.Quality = 50
	}
}
